lib/Dialect/AIE/Transforms/AIEObjectFifoStatefulTransform.cpp

```diff
@@ -967,6 +969,154 @@ struct AIEObjectFifoStatefulTransformPass
     return success();
   }
 
+  // Function that generates the IR to update runtime state of objectfifo
+  // accesses. Called by dynamicGlobalObjectFifos().
+  void updateGlobalNextIndex(OpBuilder &builder, ObjectFifoReleaseOp relOp,
+                             BufferOp globalNextIndex, arith::ConstantOp index,
+                             arith::ConstantOp size) {
+    builder.setInsertionPointAfter(relOp);
+    Value oldCounter = builder.create<memref::LoadOp>(
+        builder.getUnknownLoc(), globalNextIndex,
+        ValueRange(ArrayRef({index.getResult()})));
+    Value val = builder.create<arith::ConstantOp>(
+        oldCounter.getLoc(), builder.getIndexAttr(relOp.getSize()));
+    Value sum = builder.create<arith::AddIOp>(val.getLoc(), oldCounter, val);
+    Value newCounter = builder.create<arith::RemSIOp>(sum.getLoc(), sum, size);
+    builder.create<memref::StoreOp>(size.getLoc(), newCounter, globalNextIndex,
+                                    ValueRange(ArrayRef({index.getResult()})));
+  }
+
+  // Function that generates the IR for objectfifo accesses to be handled at
+  // runtime.
+  LogicalResult dynamicGlobalObjectFifos(DeviceOp &device, OpBuilder &builder,
+                                         std::set<TileOp> objectFifoTiles) {
+    for (auto coreOp : device.getOps<CoreOp>()) {
+      if (objectFifoTiles.count(coreOp.getTileOp()) > 0) {
```

Comment:
An early exit here can avoid indenting about one hundred lines of code:
```c++
if (objectFifoTiles.count(coreOp.getTileOp()) <= 0)
  continue;
```

Reply:
Good point! Updated.